Src/Web/Web.Shared.Net/Implementation/HttpContextExtensions.cs

```diff
@@ -37,5 +38,24 @@ public static HttpResponse GetResponse(this HttpContext context)
 
             return result;
         }
+
+        public static RequestNotification TryGetCurrentNotification(this HttpContext context)
+        {
+            RequestNotification notification = default(RequestNotification);
+            try
+            {
+                notification = context.CurrentNotification;
+            }
+            catch (NullReferenceException exp)
```

Comment:
Because the getter for CurrentNotification throws if NotificationContext is null, but it's an internal property so I can't check for it.